core/core-workflow-steps/src/test/java/org/eclipse/sw360/antenna/workflow/generators/SourceZipWriterImplTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.workflow.generators;
+
+import org.eclipse.sw360.antenna.api.IArtifactFilter;
+import org.eclipse.sw360.antenna.api.IProcessingReporter;
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceFile;
+import org.eclipse.sw360.antenna.model.reporting.MessageType;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+public class SourceZipWriterImplTest {
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    /* A bug with duplicate entries was discovered with this exemplary dependency; see Issue #601.
+       Thus, this dependency is used to test that case.
+     */
+    private static final String PATH_TO_JAR_WITH_DUPLICATE_ENTRIES = "SourceZipWriterTest/c3p0-0.9.5.3-sources.jar";
+
+    private final IArtifactFilter artifactFilterMock = mock(IArtifactFilter.class);
+    private final IProcessingReporter processingReporterMock = mock(IProcessingReporter.class);
+
+    @Test
+    public void testZipWriterDoesNotReportAnErrorWhenZippingJarWithDuplicateEntry() throws URISyntaxException {
+
+        when(artifactFilterMock.passed(any())).thenReturn(true);
+
+        final URL resource = getClass().getClassLoader().getResource(PATH_TO_JAR_WITH_DUPLICATE_ENTRIES);
+        final File file = Paths.get(resource.toURI()).toFile();
+        final String pathToJarFile = file.getAbsolutePath();
+
+        final Path pathToSourcesZip = temporaryFolder.getRoot().toPath().resolve("sources.zip");
+
+        final SourceZipWriterImpl sourceZipWriter = new SourceZipWriterImpl(pathToSourcesZip,
+                artifactFilterMock,
+                artifactFilterMock,
+                processingReporterMock);
+
+        final Artifact artifact = new Artifact();
+        final ArtifactSourceFile sourceFile = new ArtifactSourceFile(Paths.get(pathToJarFile));
+        artifact.addFact(sourceFile);
+        final List<Artifact> artifacts = Arrays.asList(artifact);
+        sourceZipWriter.createZip(artifacts);
+
+        verify(processingReporterMock, never()).add(eq(artifact),
+                eq(MessageType.PROCESSING_FAILURE),
+                contains("duplicate entry"));
+    }
```

Comment:
Not sure whether it is possible with reasonable effort, but would it be possible to check whether the duplicate entry actually exists in the target folder in the correct version?

Reply:
You are right, this should be tested. It turned out that it is not difficult to implement an will therefore be added.

Reply:
Hm, what I meant here was to check that the deflating of the archive worked correctly. So that afterwards on the local disk the file corresponding to the duplicate entry is now present with the expected content. Sorry if this was not clear before.

Reply:
What I now did, is to scan through the zip after its creation and check if the duplicate name appears (at least) twice. If I do this with java standard zip, this fails. Do you think this is not enough? I could also compare the content.

Reply:
Ah, sorry. This was a problem of my understanding. I thought that the archive was deflated and not created. So yes, the tests are fine.